### PR TITLE
Itab Exercises

### PR DESCRIPTION
**Pull request by rich-heilman**

### CODE REVIEW

**Comment by rich-heilman** [comment]
Waiting on bug fix, do not merge. Planned merge by end of August 2022.

File: exercises/practice/itab-basics/.docs/instructions.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Learn the basics about ABAP internal tables.
 
-You class has an internal table named `initial_data`.  It has three columns: `GROUP`, `NUMBER`, and `DESCRIPTION`. 
+Your class has an internal table named `initial_data`.  It has three columns: `GROUP`, `NUMBER`, and `DESCRIPTION`.
```
Comment (larshp) [inline exercises/practice/itab-basics/.docs/instructions.md:5]: remove double space, 

```suggestion
Your class has an internal table named `initial_data`. It has three columns: `GROUP`, `NUMBER`, and `DESCRIPTION`. 
```

**Comment by larshp** [comment]
all running 😅

**Comment by larshp** [comment]
Suggest adding some more unit tests, there is just one in each testclass, I'll be the first stupid developer to copy the expected values to the implementation for the most trivial solution.

At least:
* check empty input gives empty output
* one row input gives one row output(depending on the scenario)

**Comment by rich-heilman** [comment]
The linter is complaining about in-lining declarations in branch statements, but the line in fact is a LOOP statement REFERENCE INTO DATA( ).  This is NOT inside the LOOP, it is the loop.
```
  METHOD negative_count.

    DATA(actual_values) = VALUE cut->initial_numbers(
        ( group = 'A' number = 10 )
        ( group = 'B' number = 5 )
        ( group = 'A' number = 6 )
        ( group = 'C' number = 22 )
        ( group = 'A' number = 13 )
        ( group = 'C' number = 500 )
    ).

    DATA(aggregated_data) = cut->perform_aggregation( actual_values ).
    **LOOP AT aggregated_data REFERENCE INTO DATA(aggregated_item).**
      cl_abap_unit_assert=>assert_differs(
        act = aggregated_item->count
        exp = 0
      ).
    ENDLOOP.
  ENDMETHOD.
```

Can the linter be updated for this?

**Comment by larshp** [comment]
disabled, see discussion in https://github.com/abaplint/abaplint/issues/2522

**Comment by rich-heilman** [comment]
Cool, so please review and i think we may be ready to merge

**Review by larshp — APPROVED** [review]
looks good to me

**Comment by rich-heilman** [comment]
Lets go ahead and merge and make sure it all looks good.

**Pull request merged**